zerver/actions/message_edit.py

```diff
@@ -256,29 +252,24 @@ def send_message_moved_breadcrumbs(
             )
 
 
-def get_user_info_for_message_updates(message_id: int) -> MessageUpdateUserInfoResult:
-
+def get_mentions_for_message_updates(message_id: int) -> Set[int]:
     # We exclude UserMessage.flags.historical rows since those
     # users did not receive the message originally, and thus
     # probably are not relevant for reprocessed alert_words,
     # mentions and similar rendering features.  This may be a
     # decision we change in the future.
-    query = UserMessage.objects.filter(
-        message=message_id,
-        flags=~UserMessage.flags.historical,
-    ).values("user_profile_id", "flags")
-    rows = list(query)
-
-    message_user_ids = {row["user_profile_id"] for row in rows}
-
-    mask = UserMessage.flags.mentioned | UserMessage.flags.wildcard_mentioned
-
-    mention_user_ids = {row["user_profile_id"] for row in rows if int(row["flags"]) & mask}
-
-    return dict(
-        message_user_ids=message_user_ids,
-        mention_user_ids=mention_user_ids,
+    mentioned_user_ids = (
+        UserMessage.objects.filter(
+            message=message_id,
+            flags=~UserMessage.flags.historical,
+        )
+        .filter(
+            Q(flags__andnz=UserMessage.flags.mentioned)
+            | Q(flags__andnz=UserMessage.flags.wildcard_mentioned)
```

Comment:
Could this be `Q(flags__andnz=UserMessage.flags.mentioned | UserMessage.flags.wildcard_mentioned)`?

Reply:
Sure can!  Changed.